### ISSUE TITLE
XOR algo and VAGSA2 algo?

### ISSUE THREAD

**Issue by GBozkir** [jglim/UnlockECU#28]
Hello,

In the readme says that XOR algo an VAGSA2 algo is supported. I downloaded the application but don't see it. Also not in the webapplication. Something happend on it ? :-).

Brgds

**Comment by Feezex** [comment]
what is VAGSA2 ???? whuch readme?

**Comment by GBozkir** [comment]
Currently, these security providers are available:

DaimlerStandardSecurityAlgo
DaimlerStandardSecurityAlgoMod
DaimlerStandardSecurityAlgoRefG
DRVU_PROF
EDIFF290
EsLibEd25519
ESPSecurityAlgoLevel1
IC172Algo1
IC172Algo2
MarquardtSecurityAlgo
OCM172
PowertrainBoschContiSecurityAlgo1
PowertrainBoschContiSecurityAlgo2
PowertrainDelphiSecurityAlgo
PowertrainSecurityAlgo
PowertrainSecurityAlgo2
PowertrainSecurityAlgo3
PowertrainSecurityAlgoNFZ
RBTM
RDU222
RVC222_MPC222_FCW246_LRR3
SWSP177
VGSSecurityAlgo
**### VolkswagenSA2
XorAlgo**

**Comment by GBozkir** [comment]
So how to get seeds calculated for Vag ecu?

**Comment by Feezex** [comment]
add definition to db.json

**Comment by GBozkir** [comment]
What I don't understand.; is this alghortime used for all of these ecus ?

**Comment by Feezex** [comment]
Im not specialist at VAG, maybe you give us an answer

**Comment by GBozkir** [comment]
I can give you the answer if i can get add the function to db.json :-)

**Comment by Feezex** [comment]
Lets ask a boss to add it in right way.
`   public override bool GenerateKey(byte[] inSeed, byte[] outKey, int accessLevel, List<Parameter> parameters)
        {
            byte[] tape = GetParameterBytearray(parameters, "InstructionTape");
`
this may be the main trouble

**Comment by Feezex** [comment]
at OTP ? xDD

**Comment by Feezex** [comment]
read [here](http://nefariousmotorsports.com/forum/index.php?topic=4983.50;wap2)

**Comment by GBozkir** [comment]
Right.. but the problem is.. a virtual cockoit don't have a sgo or frf file.. there are only sd card flash softwares available..

**Comment by Feezex** [comment]
![directed-by-robert-weide](https://github.com/jglim/UnlockECU/assets/75809563/6776354c-bdcb-49e3-83b0-fa7ebf34eda9)

**Comment by jglim** [comment]
Sergey answers correctly [in this post](https://github.com/jglim/UnlockECU/issues/28#issuecomment-1635378404). The algos are implemented, but there are no definitions.

The VW SA2 algo is implemented from https://github.com/bri3d/sa2_seed_key. I haven't got to extracting the key parameters yet as I don't have a source for the FRF/ODX files (there's a [good guide in the same repo](https://github.com/bri3d/sa2_seed_key/issues/1#issuecomment-1314459205)). That being said, I don't think this will solve your issue as your key material isn't in the FRF/ODX.
